flink-runtime/src/test/java/org/apache/flink/runtime/io/network/buffer/BufferCompressionTest.java

```diff
@@ -0,0 +1,205 @@
+
+	@Parameters(name = "isDirect = {0}, codec = {1}, compressInPlace = {2}, decompressInPlace = {3}")
+	public static Collection<Object[]> parameters() {
+		return Arrays.asList(new Object[][] {
+			{true, "LZ4", true, false},
+			{true, "LZ4", false, true},
+			{true, "LZ4", false, false},
+			{false, "LZ4", true, false},
+			{false, "LZ4", false, true},
+			{false, "LZ4", false, false},
+		});
+	}
+
+	public BufferCompressionTest(
+			boolean isDirect,
+			String compressCodec,
+			boolean compressInPlace,
+			boolean decompressInPlace) {
+		this.compressInPlace = compressInPlace;
+		this.decompressInPlace = decompressInPlace;
+		this.compressor = new BufferCompressor(BUFFER_SIZE, compressCodec);
+		this.decompressor = new BufferDecompressor(BUFFER_SIZE, compressCodec);
+		this.bufferToCompress = createBufferAndFillWithLongValues(isDirect);
+	}
+
+	@Test
+	public void testCompressAndDecompressNetWorkBuffer() {
+		Buffer compressedBuffer = compress(compressor, bufferToCompress, compressInPlace);
+		assertTrue(compressedBuffer.isCompressed());
+
+		Buffer decompressedBuffer = decompress(decompressor, compressedBuffer, decompressInPlace);
+		assertFalse(decompressedBuffer.isCompressed());
+
+		verifyDecompressionResult(decompressedBuffer, 0, NUM_LONGS);
+	}
+
+	@Test
+	public void testCompressAndDecompressReadOnlySlicedNetworkBuffer() {
+		int offset = NUM_LONGS / 4 * 8;
+		int length = NUM_LONGS / 2 * 8;
+
+		Buffer readOnlySlicedBuffer = bufferToCompress.readOnlySlice(offset, length);
+		Buffer compressedBuffer = compress(compressor, readOnlySlicedBuffer, compressInPlace);
+		assertTrue(compressedBuffer.isCompressed());
+
+		Buffer decompressedBuffer = decompress(decompressor, compressedBuffer, decompressInPlace);
+		assertFalse(decompressedBuffer.isCompressed());
+
+		verifyDecompressionResult(decompressedBuffer, NUM_LONGS / 4, NUM_LONGS / 2);
+	}
+
+	@Test(expected = IllegalArgumentException.class)
+	public void testCompressEmptyBuffer() {
+		compress(compressor, bufferToCompress.readOnlySlice(0, 0), compressInPlace);
+	}
+
+	@Test(expected = IllegalArgumentException.class)
+	public void testDecompressEmptyBuffer() {
+		Buffer readOnlySlicedBuffer = bufferToCompress.readOnlySlice(0, 0);
+		readOnlySlicedBuffer.setCompressed(true);
+
+		decompress(decompressor, readOnlySlicedBuffer, decompressInPlace);
+	}
+
+	@Test(expected = IllegalArgumentException.class)
+	public void testCompressBufferWithNonZeroReadOffset() {
+		bufferToCompress.setReaderIndex(1);
+
+		compress(compressor, bufferToCompress, compressInPlace);
+	}
+
+	@Test(expected = IllegalArgumentException.class)
+	public void testDecompressBufferWithNonZeroReadOffset() {
+		bufferToCompress.setReaderIndex(1);
+		bufferToCompress.setCompressed(true);
+
+		decompress(decompressor, bufferToCompress, decompressInPlace);
+	}
+
+	@Test(expected = IllegalArgumentException.class)
+	public void testCompressNull() {
+		assertNull(compress(compressor, null, compressInPlace));
```

Comment:
`assertNull` would not be invoked because `compress()` would throw exception and finish the test directly. So we can remove the `assertNull`.